stingray/pulse/pulsar.py

```diff
@@ -771,9 +771,20 @@ def get_orbital_correction_from_ephemeris_file(mjdstart, mjdstop, parfile,
     toalist = _load_and_prepare_TOAs(mjds, ephem=ephem)
     m = get_model(parfile)
     delays = m.delay(toalist)
-    correction_mjd = \
+
+    correction_mjd_rough = \
         interp1d(mjds,
-                 (toalist.table['tdbld'] * units.d - delays).to(units.d).value)
+                 (toalist.table['tdbld'] * units.d - delays).to(units.d).value,
+                  fill_value="extrapolate")
+
+    def correction_mjd(mjds):
+        xvals = correction_mjd_rough.x
+        # Maybe this will be fixed if scipy/scipy#9602 is accepted
```

Comment:
What is actually wrong here in scipy?

Reply:
Scipy has no grey area: if you don't want to extrapolate, even one microsecond out of the interpolation area makes everything fail, which is unreasonable for this particular function. 
However, if you decide to extrapolate outside of the interpolation region, you're on your own, there is no way to know whether extrapolation is happening or not. What I want here is to allow extrapolation, but send a warning if it's happening.